### PR TITLE
chore(Makefile, CI): Refactor Makefile to use goreleser

### PR DESCRIPTION
**Pull request by grzesuav**

### CODE REVIEW

**Comment by grzesuav** [comment]
Also adjusted formatting to default intellij formatter

File: .goreleaser.yaml [hunk 1]
```diff
@@ -29,218 +33,200 @@ archives:
...
-  groups:
-    - title: Features
-      regexp: "^.*feat[(\\w)]*:+.*$"
-      order: 0
-    - title: 'Bug fixes'
-      regexp: "^.*fix[(\\w)]*:+.*$"
-      order: 1
-    - title: Others
-      order: 999
-  filters:
```
Comment (mikesmithgh) [inline .goreleaser.yaml:238]: 👍  This will be helpful to see all changes in the release.

**Review by mikesmithgh — APPROVED** [review]
LGTM :shipit:

File: .goreleaser.yaml (hunk 1 above)
Comment (grzesuav) [reply .goreleaser.yaml:238]: this settings are ignored, I use semantic release notes instead the ones done by `goreleaser`. However `semantic-release` configuration is not so obvious as goreleaser, so I need time to figure that out

**Pull request merged**